src/shogun/base/SGObject.h

```diff
@@ -123,6 +123,60 @@ class CSGObject
 
 	/** destructor */
 	virtual ~CSGObject();
+        
+        /** cereal serialization saver wrapper.
+         * save SGObject registered parameters to Binary archives
+         *
+         * @param filename Binary archive filename
+         */
+	void saveBinary(char* filename) const;
+        
+        /** cereal serialization saver wrapper.
+         * save SGObject registered parameters to JSON archives
+         *
+         * @param filename JSON archive filename
+         */
+	void saveJSON(char* filename) const;
+
+        /** cereal serialization saver wrapper.
+         * save SGObject registered parameters to XML archives
+         *
+         * @param filename XML archive filename
+         */
+	void saveXML(char* filename) const;
+
+        /** cereal serialization loader wrapper.
+         * load SGObject registered parameters to Binary archives
+         *
+         * @param filename Binary archive filename
+         */
+	void loadBinary(char* filename);
+        
+        /** cereal serialization loader wrapper.
+         * load SGObject registered parameters to JSON archives
+         *
+         * @param filename JSON archive filename
+         */
+	void loadJSON(char* filename);
+        
+        /** cereal serialization loader wrapper.
+         * load SGObject registered parameters to XML archives
+         *
+         * @param filename XML archive filename
+         */
+	void loadXML(char* filename);
+    
+        /** cereal serailization save method
```

Comment:
Needs to be hidden from SWIG.
Also please put more information for c++ users, and also for Shogun hackers in here